benches/cub/device/run_length_encode/basic.cu

```diff
@@ -0,0 +1,95 @@
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+#include <thrust/random.h>
+#include <thrust/fill.h>
+
+#include <tbm/range_generator.cuh>
+
+#include <cub/device/device_run_length_encode.cuh>
+
+template <typename T>
+void basic(nvbench::state &state,
+           nvbench::type_list<T>)
+{
+  const auto elements = static_cast<std::size_t>(state.get_int64("Elements"));
+
+  thrust::device_vector<T> input(elements);
+  thrust::device_vector<T> unique_output(elements);
+  thrust::device_vector<std::size_t> counts_output(elements);
+  thrust::device_vector<std::size_t> num_runs(1);
+
+  const auto pattern = state.get_string("Pattern");
+  if (pattern == "constant")
+  {
+    thrust::fill(input.begin(), input.end(), T{0});
+  }
+  else if (pattern == "sequence")
+  {
+    thrust::sequence(input.begin(), input.end());
+  }
+  else if (pattern == "random")
+  {
+    auto random_input =
+      tbm::make_range_generator<T,
+                                tbm::iterator_style::pointer,
+                                tbm::data_pattern::random>(elements);
+
+    thrust::copy(random_input.cbegin(),
```

Comment:
Might be worth doing a transform here such that `input[i] = random_input[i] % (elements / 2)` or something similar to ensure that there are a significant number of non-trivial runs to process.